File: deepseek/nivel avanzado/ruby/palindromo.rb
def es_palindromo(texto)
  limpio = texto.downcase.gsub(/[^a-z0-9]/, '')
  limpio == limpio.reverse
end

print "Ingrese una cadena de texto: "
cadena = gets.chomp
puts "¿Es palíndromo? #{es_palindromo(cadena)}"

File: deepseek/nivel avanzado/c++/palindromo.cpp
#include <iostream>
#include <algorithm>
#include <cctype>
using namespace std;

bool esPalindromo(string texto) {
    string limpio;
    for (char c : texto) {
        if (isalnum(c)) limpio += tolower(c);
    }
    string reverso = limpio;
    reverse(reverso.begin(), reverso.end());
    return limpio == reverso;
}

int main() {
    string cadena;
    cout << "Ingrese una cadena de texto: ";
    getline(cin, cadena);
    cout << "¿Es palíndromo? " << (esPalindromo(cadena) ? "Sí" : "No") << endl;
    return 0;
}